torch_xla/csrc/tensor_methods.cpp

```diff
@@ -1433,10 +1433,16 @@ XLATensor XLATensor::narrow(const XLATensor& input, xla::int64 dim,
   dim = XlaHelpers::GetCanonicalDimensionIndex(dim, input_shape.get().rank());
   xla::Shape narrow_shape = input_shape;
   narrow_shape.set_dimensions(dim, length);
-  ViewInfo view_info(std::move(narrow_shape), input_shape.get().dimensions());
+
+  ViewInfo::Type view_type = (xla::ShapeUtil::ElementsIn(input_shape) ==
```

Comment:
Is this required for correctness, could it be `ViewInfo::Type::kNarrow` in all cases? Also, I think moving this decision to a helper is justified.

Reply:
Now that we have distinct operations, it's better the caller decide the operation by spelling it out.
IMHO that is better than having some generic reshape which could narrow or reshape.
Without resize() we were already borderline with the constructor overloading. The reshape really called for it.